asteroid/data/medleydb_dataset.py

```diff
@@ -0,0 +1,243 @@
+import torch
+from torch.utils import data
+import json
+import os
+import numpy as np
+import soundfile as sf
+import random
+import torchaudio
+
+
+class MedleydbDataset(data.Dataset):
+    """MedleyDB: a dataset of annotated, royaltyfree multitrack recordings.
+
+    It provides a stereo mix and both dry and processed multitrack stems for 
+    each song in the dataset. The dataset covers a wide distribution of genres 
+    and primarily consists of full length songs with professional 
+    or nearprofessional audio quality.
+    
+    The dataset consists of 254 full lengths music tracks of
+    different genres. It provides a stereo mix and both dry and processed
+    multitrack stems for each song. 
+
+    This dataset contains a .yml object for each multitrack containining
```

Comment:
Is it a .yml or .json?

Reply:
Each multitrack in the dataset has a corresponding .yaml file (present in medleydb repo /data/metadata/) which contains information regarding each of the audio files belonging to the multitrack. The data preprocessing reads these .yaml files to find files with the desired instrument tags, and write these file paths to a .json file (along with the activity confidence array).